### DISCUSSION TITLE
Help with DALY BMS connection

### DISCUSSION THREAD

**Discussion by ismaelnoble** [ai-republic/bms-to-inverter#58] [Q&A]
Hi,
Thank you for providing us with such a useful piece of software, I hope you continue to support it in the future and folks like me having issues arent causing you much headache.
Ive been looking to connect an ecoworthy 5k (rebadged srne hyp4850s100-h) inverter to a daly smart BMS. But for the life of me i cant get past an error I am getting. I am connecting a USB-UART connector to a linux pc and have the configuration set to use the uart interface eg /dev/ttyusb0 DALY_RS485. when I startup the app logs show a connection is made but eventually fails. logs below:

![image](https://github.com/user-attachments/assets/f79ea803-0776-4a86-8bba-2c9072f16d6c)

do you have any advice for someone stuck on a particular bug.

**Comment by ai-republic** [comment]
Does your Daly only support UART or also RS485? It would be better to use a RS485 or CAN connection. I have not tried it with UART and I'm not sure if it matches the RS485 protocol. Could you dump the data coming over the UART with `cat /dev/ttyUSB0`? Then I could tell you if is the same specification.

**Comment by ismaelnoble** [comment]
The unit has an RS port but it is inactive, ive spend a few weeks confirming this. I have been able to connect to the bms using the windows application on the same usb connection on another machine with RS485 settings. running cat on the interface as requested does not show any data =(
![image](https://github.com/user-attachments/assets/95e501e6-7cf9-4048-9a43-08797b4ecdab)

ive found another thread with the same usb interface here https://github.com/ai-republic/bms-to-inverter/issues/5

**Comment by ai-republic** [comment]
Have you tried initializing your port before you start the application:
```
stty -F /dev/ttyUSB0 9600
```
?

**Comment by ismaelnoble** [comment]
noob mistake on my part. I did not, i will give it a shot when i get home
in a few hours.

**Comment by ismaelnoble** [comment]
Im reporting back that no response is coming back from the interface. I did a test using python-daly-bms on the same interface, the settings from the application show 9600, 8N1 as its default settings for daly. I do get the following from the python script;
![image](https://github.com/user-attachments/assets/c990a6dd-4f44-4f43-8f1a-f756ae00264b)

after running stty -F /dev/ttyUSB0 9600 on the machine and cat /dev/ttyUSB0 i still get no data being read. running sudo ./start.sh does the same on the system as shown below.

![image](https://github.com/user-attachments/assets/3c19f365-2d86-4cd1-8eb9-471664a427fb)

lastly thank you for taking the time to help me out.

**Comment by ai-republic** [comment]
One thing I noticed: The BMS id should be set to 1 for the Daly BMS.
Could you also attach the python_daly_bms script file? Then I can see if its using the same communication specification via UART.

**Reply by ismaelnoble** [reply]
python script from this repo https://github.com/dreadnought/python-daly-bms

**Reply by ismaelnoble** [reply]
also via the daly windows application the id is set to 1-1

**Comment by ai-republic** [comment]
Yes, but in the log file the BMS id is set to 0.
![image](https://github.com/user-attachments/assets/8ce52f4e-6279-46dd-b0f3-168a76bd2bf9)

Could you try to configure BMS id 1 in the _Configurator_ application and click _Update Configuration_?

**Comment by ismaelnoble** [comment]
Good evening, 

I am reporting back that updating the ID in the configurator application to 1 did the trick to have the bms load data in the software. I can now see the data from the web portal and in the cli logs. So we can say that this issue is solved for me. I do however now get error messages from the inverter end but I am thinking that it may be because I may have a configuration bug somewhere. I will play with it for a bit before opening another thread. 

![image](https://github.com/user-attachments/assets/46b690fe-2f42-4476-86a4-7634ee551e00)

**Comment by ai-republic** [comment]
Good to hear that your BMS is reporting back properly now! 😃 
For the exception in the log: Did you try it with the latest version from the _Configurator_ application (_Clean install_)?
I will close this thread. For the Pylon binding kindly create an issue ticket if there is a bug or you need some help on that.
